### DISCUSSION TITLE
在Popup添加了个按钮用于快速添加抓取规则

### DISCUSSION THREAD

**Discussion by jc3213** [jc3213/download_with_aria2#33] [Ideas]
![1](https://user-images.githubusercontent.com/8744936/174467146-7b7a64bc-6752-4c51-a6fc-c3941559845f.png)

@ivysrono 不知道怎样？

**Comment by jc3213** [comment]
本来只想做个指示器的，但是想了想可能有多个规则能匹配到当前网页，所以改成了这种更直观的（会显示最快匹配到的规则）。

UI方面应该还会有后续调整，现在感觉仅仅可用，不过我确实不善于制作UI，感觉多半不会好到哪去。

**Reply by escape0707** [reply]
现在的ui确实精简好用快速。其实可以往微软 metro 或者 fluent 设计那边靠一下。https://developer.microsoft.com/en-us/fluentui#/

但是只用很精简的设计反应速度也很快很棒。

主要缺一个打开文件所在文件夹位置，不过我猜是目前浏览器没给这功能是吧。

**Reply by jc3213** [reply]
打开文件所在文件夹是不被允许的。除非你使用原生`downloads`API，但是这个背离了使用aria2下载的初衷。

关于UI，下面我也提到了，我不太会做UI。最开始得时候我还只会用纯JS写UI。现在简单的HTML还好，复杂了就更愿意去依赖jQuery之类的库了。而我从最开始改写，然后到现在完全重写。基本都在用最基本的HTML跟JS命令，就是为了保证性能（毕竟很多功能我写得并不好）。

目前发现关于等待队列还有些BUG，然后就是快速访问的这个UI我考虑用之前多用户档案的思路去弄。然后可能稍微改下任务管理器的主界面，把设置，清理，新建全部挪到上面去。保证下面有足够空间去做监听跟未来的多用户档案。

**Comment by ivysrono** [comment]
先做出来试试看？
我一般都用全局抓取部分排除的模式

**Reply by jc3213** [reply]
理论上做成类似Proxy SwitchyOmega那样比较好？然而我做不了太复杂的UI，这点是个问题。另外最新的master代码已经是了。我考虑优化UI后再发布，就是目前完全没思路。

**Reply by jc3213** [reply]
顺带我把新代码移除了，先发布个修复了BUG跟优化了交互逻辑的版本。如果需要增加了新代码的版本请 [点此下载](https://github.com/jc3213/download_with_aria2/archive/530ae0b3949c78ecc7e560e35f56241cf82863df.zip)

移除代码是在这里 https://github.com/jc3213/download_with_aria2/commit/b0f9dcac8012041bf10b9d6aa3f1d8533299a736

**Reply by ivysrono** [reply]
我也不会做界面，比写逻辑难太多了。

**Comment by jc3213** [comment]
因为Background跟Popup完全是分离得所以依赖`SharedWorker`跟`WebSocket`做了现在得版本，但是`WebSocket`缺乏`aria2.onDownloadWait`事件，导致`Paused`任务变成`Waiting`时无法被正确识别，这里还要优化。aria2那边投了issue也没人维护得样子，这部分感觉得重写。

**Comment by jc3213** [comment]
感觉哪没对……就挺头疼的

![1](https://user-images.githubusercontent.com/8744936/175813363-104a82c2-0bf4-4b81-be84-96980abd7dd7.png)
